common/src/main/java/org/fao/geonet/utils/Xml.java

```diff
@@ -385,6 +385,7 @@ public static Element transform(Element xml, Path styleSheetPath, Map<String, Ob
     public static void transform(Element xml, Path styleSheetPath, OutputStream out) throws Exception {
         StreamResult resStream = new StreamResult(out);
         transform(xml, styleSheetPath, resStream, null);
+        out.close();
```

Comment:
I'm not sure if this could throw an `IllegalStateException` in the application server, since usually the `OutputStream` passed to this methods is the one from `HttpServletResponse.getOutputStream()` method.

Usually you want to close the streams in the place where they are created, and the response stream lifecycle is managed by the server, not by us.

Reply:
If you want to speed up things maybe you could use [`OutputStream.flush`](https://docs.oracle.com/javase/8/docs/api/java/io/OutputStream.html#flush--) method instead.